src/utils/fetch.ts

```diff
@@ -0,0 +1,136 @@
+import http from 'http';
+import https from 'https';
+import tls from 'tls';
+import util from 'util';
+
+import _fetch, { RequestInit } from 'node-fetch';
+
+import { getSystemCertificates } from '@/main/networking';
+
+/**
+ * CertificateVerificationError is a custom Error class that describes a TLS
+ * certificate that failed verification.
+ */
+export class CertificateVerificationError extends Error {
+  constructor(error: string, cert: tls.DetailedPeerCertificate) {
+    const wantedKeys = [
+      'subject', 'issuer', 'subjectaltname', 'valid_from', 'valid_to',
+      'fingerprint', 'fingerprint256', 'serialNumber'];
+
+    super(error);
+    this.error = error;
+    this.certChain = [];
+    while (cert) {
+      this.certChain.push(Object.fromEntries(Object.entries(cert).filter(([x]) => wantedKeys.includes(x))));
+      if (cert.issuerCertificate.fingerprint === cert.fingerprint) {
+        break;
+      }
+      cert = cert.issuerCertificate;
+    }
+  }
+
+  [util.inspect.custom](depth: number, opts: util.InspectOptionsStylized) {
```

Comment:
What is the `[util.inspect.custom]` syntax? Is this generating a method name on the fly, based on the value of `util.inspect.custom` at class loading time? That doesn't really make sense to me.